x/auth/keeper/deterministic_test.go

```diff
@@ -118,3 +152,438 @@ func pubkeyGenerator(t *rapid.T) *rapid.Generator[secp256k1.PubKey] {
+	suite.Require().NoError(err)
+
+	suite.runParamsIterations(params)
+}
+
+func (suite *DeterministicTestSuite) runAccountInfoIterations(addr sdk.AccAddress, prevRes *types.BaseAccount) {
+	for i := 0; i < 1000; i++ {
+		res, err := suite.queryClient.AccountInfo(suite.ctx, &types.QueryAccountInfoRequest{Address: addr.String()})
+		suite.Require().NoError(err)
+		suite.Require().NotNil(res)
+		suite.Require().NotNil(res.Info)
+
+		if prevRes != nil {
+			suite.Require().Equal(res.GetInfo().Address, prevRes.Address)
+			suite.Require().True(res.GetInfo().PubKey.Equal(prevRes.PubKey))
+			suite.Require().Equal(res.GetInfo().AccountNumber, prevRes.AccountNumber)
+			suite.Require().Equal(res.GetInfo().Sequence, prevRes.Sequence)
+		}
+
+		prevRes = res.GetInfo()
+	}
+}
+
+func (suite *DeterministicTestSuite) TestGRPCQueryAccountInfo() {
+	rapid.Check(suite.T(), func(t *rapid.T) {
+		accs := suite.createAndSetAccounts(t, 1)
+		suite.Require().Len(accs, 1)
+
+		acc, ok := accs[0].(*types.BaseAccount)
+		suite.Require().True(ok)
+
+		suite.runAccountInfoIterations(acc.GetAddress(), acc)
+	})
+
+	// Regression test
+	addr1, err := sdk.AccAddressFromBech32("cosmos1j364pjm8jkxxmujj0vp2xjg0y7w8tyveuamfm6")
+	suite.Require().NoError(err)
+	pub, err := hex.DecodeString("01090C02812F010C25200ED40E004105160196E801F70005070EA21603FF06001E")
+	suite.Require().NoError(err)
+
+	accNum := uint64(10087)
+	seq := uint64(10)
+
+	acc1 := types.NewBaseAccount(addr1, &secp256k1.PubKey{Key: pub}, accNum, seq)
+
+	suite.accountKeeper.SetAccount(suite.ctx, acc1)
+	suite.runAccountInfoIterations(addr1, acc1)
+}
+
+func (suite *DeterministicTestSuite) createAndReturnQueryClient(ak keeper.AccountKeeper) types.QueryClient {
+	queryHelper := baseapp.NewQueryServerTestHelper(suite.ctx, suite.encCfg.InterfaceRegistry)
+	types.RegisterQueryServer(queryHelper, ak)
+	return types.NewQueryClient(queryHelper)
+}
+
+func (suite *DeterministicTestSuite) runBech32PrefixIterations(ak keeper.AccountKeeper, preRes string) {
+	queryClient := suite.createAndReturnQueryClient(ak)
+
+	for i := 0; i < 1000; i++ {
+		res, err := queryClient.Bech32Prefix(suite.ctx, &types.Bech32PrefixRequest{})
+		suite.Require().NoError(err)
+		suite.Require().NotNil(res)
+
+		suite.Require().Equal(res.Bech32Prefix, preRes)
+		preRes = res.Bech32Prefix
+	}
+}
+
+func (suite *DeterministicTestSuite) TestGRPCQueryBech32Prefix() {
+	rapid.Check(suite.T(), func(t *rapid.T) {
+		prefix := rapid.StringMatching(`[a-zA-Z]+[1-9a-zA-Z]*`).Draw(t, "prefix")
+		ak := keeper.NewAccountKeeper(
+			suite.encCfg.Codec,
+			suite.key,
+			types.ProtoBaseAccount,
+			nil,
+			prefix,
+			types.NewModuleAddress("gov").String(),
+		)
+
+		suite.runBech32PrefixIterations(ak, prefix)
+	})
+
+	prefix := "prefix"
+	ak := keeper.NewAccountKeeper(
+		suite.encCfg.Codec,
+		suite.key,
+		types.ProtoBaseAccount,
+		nil,
+		prefix,
+		types.NewModuleAddress("gov").String(),
+	)
+
+	suite.runBech32PrefixIterations(ak, prefix)
```

Comment:
I think it's fine to use the suite's keeper for regression

```suggestion
	suite.runBech32PrefixIterations(suite.ak, "cosmos")
```